packages/client-engine-runtime/src/interpreter/serializeSql.ts

```diff
@@ -3,25 +3,36 @@ import { type ColumnType, ColumnTypeEnum, type SqlResultSet } from '@prisma/driv
 import { assertNever } from '../utils'
 
 export function serializeSql(resultSet: SqlResultSet): Record<string, unknown>[] {
-  return resultSet.rows.map((row) =>
-    row.reduce<Record<string, unknown>>((acc, value, index) => {
-      const splitByDot = resultSet.columnNames[index].split('.')
+  const mappers = resultSet.columnTypes.map((type) => {
+    switch (type) {
+      case ColumnTypeEnum.Bytes:
+        return (value: unknown) => (Array.isArray(value) ? new Uint8Array(value) : value)
+      default:
+        return (value: unknown) => value
+    }
+  })
+
+  return resultSet.rows
+    .map((row) => row.map((value, index) => mappers[index](value)))
+    .map((row) =>
+      row.reduce<Record<string, unknown>>((acc, value, index) => {
```

Comment:
Probably better to inline the first `map`:

```suggestion
    .map((row) =>
      row
        .map((value, index) => mappers[index](value)))
        .reduce<Record<string, unknown>>((acc, value, index) => {
```

The `rows` array could be large, and `map` is not lazy in JavaScript but will allocate a whole new array each time.